lib/watch-sidebar.js

```diff
@@ -6,148 +6,142 @@ import WatchView from './watch-view';
-        this.resizeSidebar = this.resizeSidebar.bind(this);
-        this.kernel = kernel;
-        this.element = document.createElement('div');
-        this.element.classList.add('hydrogen', 'watch-sidebar');
-
-        let toolbar = document.createElement('div');
-        toolbar.classList.add('toolbar', 'block');
-
-        let languageDisplay = document.createElement('div');
-        languageDisplay.classList.add('language', 'icon', 'icon-eye');
-        languageDisplay.innerText = this.kernel.kernelSpec.display_name;
-
-        let commands = document.createElement('div');
-        commands.classList.add('btn-group');
-        let removeButton = document.createElement('button');
-        removeButton.classList.add('btn', 'icon', 'icon-trashcan');
-        removeButton.onclick = () => this.removeWatch();
-        let toggleButton = document.createElement('button');
-        toggleButton.classList.add('btn', 'icon', 'icon-remove-close');
-        toggleButton.onclick = function() {
-            let editor = atom.workspace.getActiveTextEditor();
-            let editorView = atom.views.getView(editor);
-            return atom.commands.dispatch(editorView, 'hydrogen:toggle-watches');
-        };
-
-        let tooltips = new CompositeDisposable();
-        tooltips.add(atom.tooltips.add(toggleButton,
-            {title: 'Toggle Watch Sidebar'})
-        );
-        tooltips.add(atom.tooltips.add(removeButton,
-            {title: 'Remove Watch'})
-        );
-
-        this.watchesContainer = document.createElement('div');
-        _.forEach(this.watchViews, watch => {
-            return this.watchesContainer.appendChild(watch.element);
-        }
-        );
-
-        let addButton = document.createElement('button');
-        addButton.classList.add('add-watch', 'btn', 'btn-primary',
-                                 'icon', 'icon-plus', 'inline-block');
-        addButton.innerText = 'Add watch';
-        addButton.onclick = () => this.addWatch();
-
-        let resizeHandle = document.createElement('div');
-        resizeHandle.classList.add('watch-resize-handle');
-        $(resizeHandle).on('mousedown', this.resizeStarted);
-
-        toolbar.appendChild(languageDisplay);
-        toolbar.appendChild(commands);
-        commands.appendChild(removeButton);
-        commands.appendChild(toggleButton);
-
-        this.element.appendChild(toolbar);
-        this.element.appendChild(this.watchesContainer);
-        this.element.appendChild(addButton);
-        this.element.appendChild(resizeHandle);
-
-        this.kernel.addWatchCallback(() => {
-            return this.run();
-        }
-        );
-
-        this.watchViews = [];
-        this.addWatch();
-
-        this.hide();
-        atom.workspace.addRightPanel({item: this.element});
+  constructor(kernel) {
+    this.resizeStarted = this.resizeStarted.bind(this);
+    this.resizeStopped = this.resizeStopped.bind(this);
+    this.resizeSidebar = this.resizeSidebar.bind(this);
```

Comment:
They are actually necessary. I think because they are used [here](https://github.com/nteract/hydrogen/pull/503/files#diff-2809de1cb9bae13793c2862fe73ff693R118).